src/components/challenges-completion.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import Link from 'next/link';
+import React, { useState } from 'react';
+
+import { Button } from './ui/button';
+import { BarProgress } from './ui/graphElements';
+import Widget from './ui/widget';
+
+interface ChallengesCompletionProps {
+  totalChallengePercentage: number;
+  nextChallengeDate: string;
+}
+
+const ChallengesCompletion = (
+  challengesCompletionProps: ChallengesCompletionProps
```

Comment:
Destructure the props could be a good idea here

```suggestion
  { totalChallengePercentage, nextChallengeDate }: ChallengesCompletionProps
```